### PR TITLE
feat(strategy-tests): add extra_keys field for StartIdentities and use random identities for transfers

### PR DESCRIPTION
**Pull request by pauldelucia**
## Issue being fixed or feature implemented

We might want Strategy StartIdentities to have transfer keys or other extra keys in addition to the main keys. Also, there was an issue with public key IDs in `transitions.rs`.

## What was done?

We add an extra_keys field similar to that in IdentityInserts and fix the issue in `transitions.rs`.

## How Has This Been Tested?

Platform TUI strategy tests

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/strategy-tests/src/lib.rs [hunk 1]
```diff
@@ -1328,10 +1330,25 @@ impl Strategy {
...
+            )]
+            .into();
+            let mut new_key_count = self.start_identities.keys_per_identity - 1;
+
+            // Also an authentication key with critical security level for documents
+            key_maps.insert(
+                Purpose::AUTHENTICATION,
+                [(SecurityLevel::CRITICAL, vec![KeyType::ECDSA_SECP256K1])].into(),
+            );
+            new_key_count -= 1;
```
Comment (QuantumExplorer) [inline packages/strategy-tests/src/lib.rs:1346]: I on purpose didn't want to add the transfer key to all identities.

File: packages/strategy-tests/src/lib.rs [hunk 2]
```diff
@@ -418,7 +418,9 @@ impl Strategy {
         let (identities, mut state_transitions): (Vec<Identity>, Vec<StateTransition>) =
             identity_state_transitions.into_iter().unzip();
 
-        // Do we also need to add identities to the identity_nonce_counter?
+        for identity in &identities {
+            identity_nonce_counter.insert(identity.id(), 1);
+        }
```
Comment (QuantumExplorer) [inline packages/strategy-tests/src/lib.rs:423]: I think this means we would start things at 2.

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

File: packages/strategy-tests/src/lib.rs (hunk 2 above)
Comment (pauldelucia) [reply packages/strategy-tests/src/lib.rs]: Fixed (removed it because it wasn't necessary)

File: packages/strategy-tests/src/lib.rs (hunk 1 above)
Comment (pauldelucia) [reply packages/strategy-tests/src/lib.rs]: Fixed. I added extra_keys field for StartIdentities struct and the users can select whether or not to add a transfer key in the TUI

File: packages/strategy-tests/src/lib.rs [hunk 3]
```diff
@@ -1072,23 +1071,41 @@ impl Strategy {
...
-                        // Recipient is the second in the list
-                        let recipient = &identities_clone[1];
                         for _ in 0..count {
+                            let identities_count = current_identities.len();
+                            if identities_count == 0 {
+                                break;
+                            }
+
+                            // Select a random identity from the current_identities for the sender
+                            let random_index_sender = thread_rng().gen_range(0..identities_count);
```
Comment (QuantumExplorer) [inline packages/strategy-tests/src/lib.rs:1081]: don't you want to use rng, so that tests are determinisitic based on the seed?

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

File: packages/strategy-tests/src/lib.rs (hunk 3 above)
Comment (pauldelucia) [reply packages/strategy-tests/src/lib.rs]: Fixed. Changed to rng

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**